src/test/java/com/amazon/dlic/auth/http/jwt/keybyoidc/HTTPJwtKeyByOpenIdConnectAuthenticatorTest.java

```diff
@@ -108,20 +111,94 @@ public void testRoles() throws Exception {
 		Assert.assertEquals(TestJwts.TEST_ROLES, creds.getBackendRoles());
 	}
 
+
 	@Test
 	public void testExp() throws Exception {
 		Settings settings = Settings.builder().put("openid_connect_url", mockIdpServer.getDiscoverUri()).build();
 
 		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
-
+		
 		AuthCredentials creds = jwtAuth.extractCredentials(
 				new FakeRestRequest(ImmutableMap.of("Authorization", TestJwts.MC_COY_EXPIRED_SIGNED_OCT_1),
 						new HashMap<String, String>()),
 				null);
 
 		Assert.assertNull(creds);
+	}	
+
+	@Test
+	public void testExpInSkew() throws Exception {
+		Settings settings = Settings.builder()
+			.put("openid_connect_url", mockIdpServer.getDiscoverUri())
+			.put("clockSkewToleranceSeconds", "10")
+			.build();
+
+		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
+
+		long expiringDate = System.currentTimeMillis()/1000-5;
+
+		JwtToken jwt_token = TestJwts.create(TestJwts.MCCOY_SUBJECT, TestJwts.TEST_AUDIENCE, 
+		TestJwts.ROLES_CLAIM, TestJwts.TEST_ROLES_STRING, 
+		JwtConstants.CLAIM_EXPIRY, expiringDate);
+
+		String token=TestJwts.createSigned(jwt_token, TestJwk.OCT_1);
+		AuthCredentials creds = jwtAuth.extractCredentials(
+				new FakeRestRequest(ImmutableMap.of("Authorization", "bearer "+token),
+						new HashMap<String, String>()),
+				null);
+
+		Assert.assertNotNull(creds);
 	}
 
+	@Test
+	public void testNbf() throws Exception {
+		Settings settings = Settings.builder()
+			.put("openid_connect_url", mockIdpServer.getDiscoverUri())
+			.put("clockSkewToleranceSeconds", "0")
+			.build();
+
+		HTTPJwtKeyByOpenIdConnectAuthenticator jwtAuth = new HTTPJwtKeyByOpenIdConnectAuthenticator(settings, null);
+
+		long nbf = 5+System.currentTimeMillis()/1000;
+
+		JwtToken jwt_token = TestJwts.create(TestJwts.MCCOY_SUBJECT, TestJwts.TEST_AUDIENCE, 
+		TestJwts.ROLES_CLAIM, TestJwts.TEST_ROLES_STRING, 
+		JwtConstants.CLAIM_NOT_BEFORE, nbf);
+
+		String token=TestJwts.createSigned(jwt_token, TestJwk.OCT_1);
+		AuthCredentials creds = jwtAuth.extractCredentials(
+				new FakeRestRequest(ImmutableMap.of("Authorization", "bearer "+token),
+						new HashMap<String, String>()),
+				null);
+
+		Assert.assertNull(creds);
+	}
+
+	@Test
```

Comment:
See above. I don't know what the best name for the tests would be but consider something that gives you an idea what they do.

Reply:
see above